test/e2e/apps/replica_set.go

```diff
@@ -412,3 +416,63 @@ func testRSScaleSubresources(f *framework.Framework) {
 	framework.ExpectEqual(*(rs.Spec.Replicas), int32(4), "replicaset should have 4 replicas")
 
 }
+
+// ReplicaSet Replace and Patch tests
+func testRSLifeCycle(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+	zero := int64(0)
+
+	// Create webserver pods.
+	rsPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  WebserverImageName,
+	}
+
+	rsName := "test-rs"
+	replicas := int32(1)
+	rsPatchReplicas := int32(3)
+	rsPatchImage := imageutils.GetE2EImage(imageutils.Pause)
+
+	// Create a ReplicaSet
+	rs := newRS(rsName, replicas, rsPodLabels, WebserverImageName, WebserverImage, nil)
+	_, err := c.AppsV1().ReplicaSets(ns).Create(context.TODO(), rs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	// Verify that the required pods have come up.
+	err = e2epod.VerifyPodsRunning(c, ns, "sample-pod", false, replicas)
+	framework.ExpectNoError(err, "error in waiting for pods to come up: %s", err)
```

Comment:
better message? we are logging an error during verification ( we are not waiting per se )

Reply:
You are correct, Thank you.
Updated with a better message.